internal/controllers/pool/controller.go

```diff
@@ -113,15 +118,18 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	if !pool.ObjectMeta.DeletionTimestamp.IsZero() {
 		// Attempt to delete pool on C2
 		if err := r.api.DeletePool(ctx, req.NamespacedName); err != nil {
-			if strings.Contains(err.Error(), "cannot delete pool: volumes exist in pool") {
+			if err == storageos.ErrPoolNotFound {
+				// nothing to do here
+			} else if strings.Contains(err.Error(), "cannot delete pool: volumes exist in pool") {
 				// Pool on C2 is being used by a volume and cannot be deleted at this time.
 				r.recorder.Event(pool, "Warning", "Cannot Delete Pool", "This pool is in use by a volume.")
-				// Reset deletion timestamp and return no error as we do not want to re-reconcile.
-				pool.ObjectMeta.DeletionTimestamp.Reset()
-				return ctrl.Result{}, r.Client.Update(ctx, pool)
+				r.Client.Update(ctx, pool)
```

Comment:
I don't think we need this update call now if we're no longer resetting the timestamp?

Reply:
I didn't know if we needed to call update to add the event